src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -137,163 +132,173 @@ public void clearCache() {
   }
 
   /**
-   * Deserialize the query context from the base64 encoded arrow data.
-   *
-   * @param data the base64 encoded query context that was serialized using Arrow.
-   */
-  public void deserializeFromArrowBase64(String data) {
-    synchronized (this) {
-      logger.debug("deserializeFromArrowBase64() called: data from the server = {}", data);
+  * @param data: the QueryContext Object serialized as a JSON format string
+  */
+  public void deserializeQueryContextJson(String data) {
 
+    synchronized (this) {
       // Log existing cache entries
       logCacheEntries();
 
       if (data == null || data.length() == 0) {
         // Clear the cache
         clearCache();
 
-        logger.debug("deserializeFromArrowBase64() returns");
         // Log existing cache entries
         logCacheEntries();
```

Comment:
What's the point of logging if we've already cleared the cache? Should the positions be reversed?

Reply:
Because Thierry mentioned that `don't log in a synchronize block`, so I removed the logs in the synchronize blocks. Should I move them back?

Reply:
I think my question was more for @sfc-gh-igarish why are we calling function on line 151 after clearing cache on line 148 it doesn't make sense.